### ISSUE TITLE
Bad query with GraphQL when using computed collection in entity

### ISSUE THREAD

**Issue by Brewal** [api-platform/api-platform#1748]
**API Platform version(s) affected**: 2.5.7

**Description**  

When defining an attribute in an entity that is not mapped with the doctrine ORM, the collection is well defined in the generated GraphQL documentation, but if it is requested in a GraphQL query, a `QueryException` occurs. 
Note : there is no problem when using the corresponding REST endpoint. It also works with a single computed attribute relation (not a collection)

**How to reproduce**  
Entities : 

```php
/**
 * @ORM\Entity
 */
class FooEntity {
    /**
     * @ORM\Id
     * @ORM\Column(type="uuid_binary_ordered_time", unique=true)
     * @ORM\GeneratedValue(strategy="CUSTOM")
     * @ORM\CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidOrderedTimeGenerator")
     */
    public $id;

    /**
     * The comment line bellow makes the collection discoverable by GraphQL
     * @var Collection<BarEntity>
     */
    private $bars; // not mapped with ORM

    ...

    public function __construct()
    {
        $this->bars = new ArrayCollection();
    }
}

/**
 * @ORM\Entity
 */
class BarEntity {
    /**
     * @ORM\Id
     * @ORM\Column(type="uuid_binary_ordered_time", unique=true)
     * @ORM\GeneratedValue(strategy="CUSTOM")
     * @ORM\CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidOrderedTimeGenerator")
     */
    public $id;

    /**
     * @ORM\Column(type="string")
     */
    private $barAttribute;
}
```
Query : 

```graphql
foos  {
  edges {
    node {
      bars {
        edges {
          node {
            barAttribute
          }
       }
    }
}
```
Error : 
> [Syntax Error] line 0, col 66: Error: Unexpected 'WHERE'
> Doctrine\ORM\Query\
QueryException
SELECT o FROM App\Entity\BarEntity o WHERE o IN(SELECT WHERE id_a1.id = :id_p1) ORDER BY o.id ASC

**Comment by alanpoulain** [comment]
Yes it is not supported yet but PR are very welcome :slightly_smiling_face:

**Comment by alanpoulain** [comment]
Related: https://github.com/api-platform/core/issues/1870

**Comment by Brewal** [comment]
@alanpoulain Ok thanks ! Maybe you could quickly point me out where I should make some changes to the code ? I'll be happy to make a PR if I come up with a clean fix

**Comment by alanpoulain** [comment]
I think it's not a fix but more like a feature (the related class needs to be introspected like a resource).
You can see the "issue" here:
https://github.com/api-platform/core/blob/8711c800bbbb33d60ad6bbce8c3805b21a9d1fcc/src/GraphQl/Type/TypeConverter.php#L119
There are already some Behat tests too:
https://github.com/api-platform/core/blob/master/features/graphql/non_resource.feature
You can also have some inspiration in the JsonSchemaFactory which seems to handle them:
https://github.com/api-platform/core/blob/8711c800bbbb33d60ad6bbce8c3805b21a9d1fcc/src/JsonSchema/SchemaFactory.php#L70

**Comment by soyuka** [comment]
Closing as stale per #2987 (no activity since 2021-01-05). This repository is transitioning to a CLI installer. If this still reproduces against a current release of API Platform, please open a new issue on `api-platform/core` with a minimal reproduction.
